### PR TITLE
Extract CMake project type indication logic

### PR DESCRIPTION
**Pull request by rita-sonarsource**
Part of #2198

### CODE REVIEW

File: src/Core/CFamily/ICFamilyProjectTypeIndicator.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarLint.VisualStudio.Core.CFamily
+{
+    public interface ICFamilyProjectTypeIndicator
+    {
+        bool IsCMake();
```
Comment (rita-sonarsource) [inline src/Core/CFamily/ICFamilyProjectTypeIndicator.cs:25]: Not sure yet about the API, but for now this is what I need for telemetry as well.
I will probably add another "IsVcx()" method for telemetry.

Comment (duncanp-sonar) [reply src/Core/CFamily/ICFamilyProjectTypeIndicator.cs:25]: Missing doc comments for the interface and method.
What's the expected behaviour if there isn't an open solution - false?

File: src/Integration.UnitTests/ProjectToLanguageMapperTests.cs [hunk 2]
```diff
@@ -90,36 +87,27 @@ public void GetAllBindingLanguagesForProject_SupportedProjectType_CPP_ReturnsCor
...
-        [DataRow("c:\\some directory\\sub\\CMakeLists.txt")]
-        [DataRow("c:\\some directory\\sub\\folder\\CMakeLists.txt")]
-        public void GetAllBindingLanguagesForProject_OpenAsFolderProject_HasCmakeFiles_CFamilyLanguages(string cmakeListsLocation)
+        public void GetAllBindingLanguagesForProject_CMakeProject_CFamilyLanguages()
         {
-            var folderWorkspaceService = SetupOpenAsFolder("c:\\some directory");
-            var fileSystem = new MockFileSystem();
-            fileSystem.AddDirectory("c:\\some directory");
-            fileSystem.AddFile(cmakeListsLocation, new MockFileData(""));
+            var folderWorkspaceService = SetupCMakeProject(true);
```
Comment (duncanp-sonar) [inline src/Integration.UnitTests/ProjectToLanguageMapperTests.cs:92]: The variable name 'folderWorkspaceService' needs to be updated in all of the tests in this class.

**Review by duncanp-sonar — APPROVED** [review]
Looks good, just comments about comments and out-of-date variable names.

**Pull request merged**